tools/builder/check_code_format.sh

```
@@ -0,0 +1,45 @@
+
+set -e -o pipefail
+
+FILES_TO_CHECK=$(git diff --name-only HEAD^ | (grep -E ".*\.(cpp|cc|c\+\+|cxx|c|h|hpp|inl|java|js)$" || true) \
+                                            | (grep -v "^src/thirdparty/.*/.*" || true))
+
+if [ -z "$FILES_TO_CHECK" ]; then
+  echo "There is no source code to check the formatting."
+  exit 0
+fi
+
+if FORMAT_DIFF=$(git diff -U0 HEAD^ -- $FILES_TO_CHECK | clang-format-diff -p1 -style=file) && [ -z "$FORMAT_DIFF" ]; then
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)